src/tensor/domain.ts

```diff
@@ -0,0 +1,119 @@
+const EMOJI = "Emoji";
+
+export const DOMAIN_MCC_LIST = [
+  "E5ZnBpH9DYcxRkumKdS4ayJ3Ftb6o3E8wSbXw4N92GWg",
+  "86deDknZeDhko46gB8SqK7rYc5HnSBjKDvo6Mi7viYS9",
+  "6bsj8ybPa9xsc6pcAme4x6LvhKvtCmgA4TwwG4qtFw5Z",
+  "GYLiNNu4pqL6QvZKYHW2EMoibVFm2aVJsPHpUVLcU6pL",
+  "7yQYe84W7a5VgNvtRzsvy7mPRed5gmL9HnvJfsbPWK9J",
+];
+
+export const languageRegex = {
+  English: /[a-zA-Z]/,
+  Arabic: /[\u0600-\u06FF]/,
+  Chinese: /[\u4e00-\u9FFF]/,
+  Cyrillic: /[\u0400-\u04FF]/,
+  Hindi: /[\u0900-\u097F]/,
+  Japanese: /[\u3040-\u309F\u30A0-\u30FF]/,
+  Korean: /[\uAC00-\uD7AF]/,
+};
+
+// Use this when emojis are in strings because emojis aren't always the same size
+const visibleLength = (str: string) => [...new Intl.Segmenter().segment(str)].length;
+
+const isNumeric = (str: string) => /^\d+$/.test(str);
+const isAlpha = (str: string) => /^[a-zA-Z]+$/.test(str);
+const emojiLength = (str: string) =>
+  [...new Intl.Segmenter().segment(str)]
+    .map((emoji) => emoji.segment)
+    .filter((char) => /\p{Extended_Pictographic}/gu.test(char)).length;
+
+// If performance of this sucks, we can make a native addon for this
+const isPalindrome = (str: string) => str === str.split("").reverse().join("");
+
+const determineNumberClub = (name: string) => {
+  const number = parseInt(name);
+  if (number <= 1000) {
+    return "1k Club";
+  } else if (number <= 10000) {
+    return "10k Club";
+  } else if (number <= 100000) {
+    return "100k Club";
+  }
+};
+
+const classifyDomainAttributes = (
+  name: string,
+  attributes: {
+    trait_type: string;
+    value: string;
+  }[]
+) => {
+  let newAttributes: Attribute[] = [];
+
+  // Knowing there's no standard, formatting like this reduces chance of duplicates in the future
+  const currentTraits = new Set(attributes.map((trait) => trait.trait_type));
+
+  const letterCount = visibleLength(name);
+  const letterKey = `${letterCount}letter${letterCount === 1 ? "" : "s"}`;
+
+  if (isNumeric(name)) {
+    // Digits only trait
+    if (!currentTraits.has(DIGITS_ONLY)) {
+      newAttributes.push({ trait_type: DIGITS_ONLY, value: "true" });
+    }
+    // Number club traits
+    const numberClub = determineNumberClub(name);
+    if (numberClub && !currentTraits.has(numberClub)) {
+      newAttributes.push({ trait_type: numberClub, value: "true" });
+    }
+  } else if (isAlpha(name) && !currentTraits.has(LETTERS_ONLY)) {
+    // Letters only trait
+    newAttributes.push({ trait_type: LETTERS_ONLY, value: "true" });
+  } else if (emojiLength(name) > 0 && !currentTraits.has(EMOJI)) {
+    // Emoji trait
+    newAttributes.push({ trait_type: EMOJI, value: "true" });
+  }
+
+  // Letter count trait
+  if (!currentTraits.has(letterKey) && !currentTraits.has(letterKey)) {
+    newAttributes.push({
+      trait_type: `${letterCount} Letter${letterCount === 1 ? "" : "s"}`,
+      value: "true",
+    });
+  }
+
+  // Palindrome trait
+  if (!currentTraits.has(PALINDROME) && isPalindrome(name)) {
```

Comment:
Maybe u can do this as a concat of a bunch of .filter().map() calls? Not sure if that is more/less efficient than pushing. It would be a bit nicer tho.

Reply:
yeahh was thinking of that too, another option i had was something like this:
<img width="576" alt="Screenshot 2024-02-02 at 4 40 54 PM" src="https://github.com/tensor-hq/tensor-common/assets/54694188/f7a3bf0b-8fcd-4e67-88f3-37818837b977">

IDK if its ok and what the "🤮" was referring to lol